test/parallel/test-http-client-reject-unexpected-agent.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const http = require('http');
+
+const baseOptions = {
+  method: 'GET',
+  port: undefined,
+  host: '127.0.0.1',
+};
+
+const failingAgentOptions = [
+  true,
+  'agent',
+  {},
+  1,
+  () => null,
+  Symbol(),
+];
+
+const acceptableAgentOptions = [
+  false,
+  undefined,
+  null,
+  new http.Agent(),
+];
+
+const server = http.createServer((req, res) => {
+  res.end('hello');
+});
+
+let numberOfResponses = 0;
```

Comment:
the numberOfResponses is really only used to close the server. The callback for the .on('response') listener is wrapped in common.mustCall() and that is called once per request, so if any fail the whole thing should fail. Is that acceptable or do we need to also count the responses on exit?